src/Core/Managed/Net40/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -252,7 +252,7 @@ private static void InitializeComponents(TelemetryConfiguration configuration)
             InitializeComponent(configuration.TelemetryChannel, configuration);            
             InitializeComponents(configuration.TelemetryInitializers, configuration);
             InitializeComponents(configuration.ContextInitializers, configuration);
-
+            InitializeComponents(configuration.TelemetryProcessors.TelemetryProcessors, configuration);
```

Comment:
I guess TelemetryProcessorChain should implement ITelemetryModule and in the initialize method, initialize the ones that implement ITelemetryModule